x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -72,6 +72,12 @@ export function generateMappings(fields: Field[]): IndexTemplateMappings {
       switch (type) {
         case 'group':
           fieldProps = generateMappings(field.fields!);
+          attemptAddDynamicAndEnabled(fieldProps, field);
```

Comment:
Could we not mutate the input parameter `fieldProps` in these calls? In other places we return a copy, so that line would read
```
fieldProps = attemptAddDynamicAndEnabled(fieldProps, field);
```
and it is more readable when we stay consistent throughout the code base.

Reply:
good point, I updated to keep it similar with the other cases.